### DISCUSSION TITLE
Enable ffmpeg with h264_v4l2m2m encoder

### DISCUSSION THREAD

**Discussion by dakhouya** [microsoft/vcpkg#39223] [Q&A] [answered]
Hi, 

Using master branch of vcpkg, I'm trying to build ffmpeg libraries and enable the h264_v4l2m2m encoder. 
To do so I understand that the only way right now is to use the autodetection which is set to disable here
https://github.com/microsoft/vcpkg/blob/b27651341123a59f7187b42ef2bc476284afb310/ports/ffmpeg/portfile.cmake#L31

My solution right now is to create my own overlay-port with ffmpeg port so I remove the `--disable-autodetect` myself.

So I have two questions
- Is there a better solution when using vcpkg? 
- Would a feature to enable or disable the autodetection would be a good feature to add the the port?

**Comment by dg0yt** [comment, marked as answer]
> * Would a feature to enable or disable the autodetection would be a good feature to add the the port?

No.
For binary artifacts caching in vcpkg, it is necessary to know the dependencies in advance. Autodetection changes the actual dependencies based on what happens to be installed, turning re-use of the cached artifact into a lottery.


> * Is there a better solution when using vcpkg?

An overlay port is a suitable approach for this type of modification of ffmpeg. That port doesn't use standard maintainer functions which alternatively offer triplet variables for customization. 
And I would recommend to look for a specific configuration parameter instead of turning on autodetection.

**Reply by dakhouya** [reply]
Thanks for the quick answer! 
I'll dig a bit more in ffmpeg to see if there's a way to enable it through configuration.
